### PR TITLE
Bridge/PhpUnit: Fix check for project directory

### LINKED ISSUE #44863: phpunit-bridge: Doesn't work properly with vendor being a symlink
### Symfony version(s) affected

<=6.0.0

### Description

When `./vendor` is a symlink, simple-phpunit emits some warning and then installs PHPUnit to `./phpunit-8.5-0` every time being invoked.

```
$ ./vendor/bin/simple-phpunit tests/EncoderSelectionTest.php
PHP Warning:  chdir(): No such file or directory (errno 2) in /Users/glen/Library/Caches/glen/composer/dvideo-admin/vendor/symfony/phpunit-bridge/bin/simple-phpunit.php on line 178

Warning: chdir(): No such file or directory (errno 2) in /Users/glen/Library/Caches/glen/composer/dvideo-admin/vendor/symfony/phpunit-bridge/bin/simple-phpunit.php on line 178
The "hirak/prestissimo" plugin (installed globally) was skipped because it requires a Plugin API version ("^1.0.0") that does not match your Composer installation ("2.1.0"). You may need to run composer update with the "--no-plugins" option.
Creating a "phpunit/phpunit" project at "./phpunit-8.5-0"
                                                                       Installing phpunit/phpunit (8.5.22)
Plugins have been disabled.
  - Installing phpunit/phpunit (8.5.22): Extracting archive
Created project in /Users/glen/OneDrive - Foo Ltd/scm/dvideo/dvideo-admin/phpunit-8.5-0
...
```

### How to reproduce

```
composer require symfony/phpunit-bridge --dev
mv vendor /tmp
ln -s /tmp/vendor
./vendor/bin/simple-phpunit
./vendor/bin/simple-phpunit
```

### Possible Solution

Fall back to use `getcwd()/composer.json` if `dirname()` traversal fails. At least fix bug that PHPUnit is installed every invocation.

### Additional Context

I keep my working dirs in OneDrive, excluding `vendor`. OneDrive doesn't support excluding filenames or directories from sync, hence the symlink hack.

### PR DESCRIPTION
**Pull request by glensc**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #44863
| License       | MIT

### CODE REVIEW

**Comment by glensc** [comment]
Targeted 4.4 branch according to:
- https://symfony.com/doc/6.0/contributing/code/pull_requests.html#choose-the-right-branch

File: src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php [hunk 1]
```diff
@@ -118,7 +118,7 @@
 $COMPOSER_JSON = getenv('COMPOSER') ?: 'composer.json';
 
 $root = __DIR__;
-while (!file_exists($root.'/'.$COMPOSER_JSON) || file_exists($root.'/DeprecationErrorHandler.php')) {
+while (!file_exists($root.'/'.$COMPOSER_JSON) || !file_exists($root.'/DeprecationErrorHandler.php')) {
```
Comment (glensc) [inline src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php:121]: I don't understand this code, why the whole loop can't be just replaced with:

```php
$root = dirname(__DIR__);
```

the root needs to be a directory of `symfony/phpunit-bridge`, so the `DeprecationErrorHandler.php` is in that directory, right?

Comment (glensc) [reply src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php:121]: The DeprecationErrorHandler check was added in:
- https://github.com/symfony/phpunit-bridge/commit/a558e6a6a754297c8aacd2f179cec54fe343ca4d

Comment (nicolas-grekas) [reply src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php:121]: Here, we're traversing $root to find the parent dir that contains the composer.json file we're looking for. The second file_exists check ensure we don't stop at the composer.json file of the bridge itself.

**Comment by nicolas-grekas** [comment]
Closing as this is not the correct patch. Please send another try when you're ready.

File: src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php (hunk 1 above)
Comment (glensc) [reply src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php:121]: Yeah, I can see what the code does, but why is all that needed? what's wrong with my proposal:
- https://github.com/symfony/symfony/pull/45930#discussion_r841259869

what's the dynamic part needed? `vendor/symfony/phpunit-bridge` structure is always consistent, i.e what is in git repo.

**Pull request closed without merging**